caiman/source_extraction/cnmf/cnmf.py

```diff
@@ -27,22 +26,22 @@
 import scipy
 import sys
 
-from .estimates import Estimates
-from .initialization import initialize_components, compute_W
-from .map_reduce import run_CNMF_patches
-from .merging import merge_components
-from .params import CNMFParams
-from .pre_processing import preprocess_data
-from .spatial import update_spatial_components
-from .temporal import update_temporal_components, constrained_foopsi_parallel
-from .utilities import update_order
-from ... import mmapping
-from ...components_evaluation import estimate_components_quality
-from ...motion_correction import MotionCorrect
-from ...utils.utils import save_dict_to_hdf5, load_dict_from_hdf5
-from caiman import summary_images
-from caiman import cluster
+import caiman
+from caiman.components_evaluation import estimate_components_quality
+import caiman.mmapping
+from caiman.motion_correction import MotionCorrect
 import caiman.paths
+from caiman.source_extraction.cnmf.estimates import Estimates
+from caiman.source_extraction.cnmf.initialization import initialize_components, compute_W
+from caiman.source_extraction.cnmf.map_reduce import run_CNMF_patches
+from caiman.source_extraction.cnmf.merging import merge_components
+from caiman.source_extraction.cnmf.params import CNMFParams
+from caiman.source_extraction.cnmf.pre_processing import preprocess_data
+from caiman.source_extraction.cnmf.spatial import update_spatial_components
+from caiman.source_extraction.cnmf.temporal import update_temporal_components, constrained_foopsi_parallel
```

Comment:
These I like. I won't have to read through a bunch of dots to get to the method/function name in the code. 

Some of the other ones (e.g., in `movies.py`) I worry the readability suffers.

Reply:
I'm having trouble coming up with a general rule for this. Sometimes I want a dot, maybe even two. Often I want none. It is probably impossible to come up with a general algorithm based on length of variables or number of dots, because it has to do with what seems "natural" after spending a lot of time reading the code. These ones seems natural. (Some) of the ones in `movies.py` seem less natural and a bit more like they were done by a refactoring algorithm (not saying they were but you get my point I hope). 

I'll take a bit of time in `movies.py` to illustrate what I'm thinking.